pymatgen/analysis/phase_diagram.py

```diff
@@ -139,21 +115,47 @@ def __init__(self, entry, chempots, name=None):
             name: Optional parameter to name the entry. Defaults to the reduced
                 chemical formula of the original entry.
         """
-        comp = entry.composition
+        super().__init__(
```

Comment:
This `__init__` has changed substantially, what connection does this have to the `__hash__` and `__eq__`?

Reply:
This change was to introduce the `_composition` with `composition` property pattern for consistency. I went for this pattern after we concluded that having them be immutable wasn't sensible.